pkg/utils/test/oc/runner.go

```diff
@@ -0,0 +1,174 @@
+package oc
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	osexec "os/exec"
+	"strings"
+	"time"
+
+	//logger "github.com/ViaQ/logerr/v2/log"
+	//log "github.com/ViaQ/logerr/v2/log/static"
+	"github.com/go-logr/logr"
+)
+
+var log logr.Logger
+
+// Runner is for executing the command. It provides implementation for
+// the methods in oc.Command interface.
+// Other commands like oc.Exec, oc.Get, oc.Literal collect their arguments
+// and use Runner to run the commad with arguments.
+// It provides different modes of executing the commands, Run/RunFor/Output/OutputFor
+//
+// As fas as possible, it is to be kept independent of oc command syntax.
+// TODO(vimalk78)
+// Move KUBECONFIG out from here
+
+// CMD is the command to be run by the runner
+const CMD string = "oc"
+
+// runner encapsulates os/exec/Cmd, collects args, and runs CMD
+type runner struct {
+	*osexec.Cmd
+
+	args       []string
+	configPath string
+
+	// This must be set by oc.Commands to collect arguments before calling Run
+	collectArgsFunc func() []string
+
+	tostdout bool
```

Comment:
```suggestion
	toStdout bool
```
I feel it is better to set `output io.Writer` so that you can pass in any io.Writer including `os.Stdout and `Stderr`

Reply:
this package is copied as-it-is from logging (path mentioned in commit message), can we treat it as a 3rd part code used only for test, and not do a line by line review?

Reply:
changed